### PR TITLE
For #18853: Add metrics for default browser new tab experiment.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Note: needs https://github.com/mozilla-mobile/fenix/pull/18872 to land to add trigger for metrics tracking.

### CODE REVIEW

**Comment by mcarare** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?
This will help us understand user interaction with the "set as default browser" card on the new tab page in the default browser experiment.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
Determine how many people interact directly with our message.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?
No other alternative.

4) Can current instrumentation answer these questions?
No.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki.   

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
User clicks on set default browser | category 1 - technical data | #18853 |
User clicks the close button | category 1 - technical data | #18853 |

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
[https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md](https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md)

7) How long will this data be collected?  Choose one of the following:
This is scoped to a time-limited experiment/project until date 09-01-2021 or earlier, when the experiment is removed.

8) What populations will you measure?
The audience is controlled by a Nimbus experiment which will probably limit the reach.

9) If this data collection is default on, what is the opt-out mechanism for users?
Default Glean SDK opt-out mechanism.

10) Please provide a general description of how you will analyze this data.
Glean / Amplitude

11) Where do you intend to share the results of your analysis?
Only on Glean, Amplitude and with mobile teams.

12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
 No third-party tool will use this data.

**Comment by mcarare** [comment]
tagging @tdsmith for data review.

File: app/metrics.yaml [hunk 1]
```diff
@@ -5201,3 +5201,31 @@ android_keystore_experiment:
...
+  set_default_browser_clicked:
+    type: event
+    description: |
+      Set default browser was clicked from new tab screen.
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/18853
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/18895
+    data_sensitivity:
+      - technical
```
Comment (tdsmith) [inline app/metrics.yaml:5215]:
```suggestion
      - interaction
```

File: app/metrics.yaml [hunk 2]
```diff
@@ -5201,3 +5201,31 @@ android_keystore_experiment:
...
+  close_experiment_card_clicked:
+    type: event
+    description: |
+      Close experiment card was clicked from new tab screen.
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/18853
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/18895
+    data_sensitivity:
+      - technical
```
Comment (tdsmith) [inline app/metrics.yaml:5228]:
```suggestion
      - interaction
```

**Review by tdsmith — COMMENTED** [review]
data-review+

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?

Yes, in the probe definition files and the [Glean Dictionary](https://dictionary.protosaurus.dev/).

2) Is there a control mechanism that allows the user to turn the data collection on and off?

Yes, the Firefox telemetry opt-out.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

n/a

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Cat 2, interaction data.

5) Is the data collection request for default-on or default-off?

Default-on.

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No.

7) Is the data collection covered by the existing Firefox privacy notice?

Yes.

8) Does there need to be a check-in in the future to determine whether to renew the data?

Interested teams are responsible for renewing the collection before it expires.

9) Does the data collection use a third-party collection tool?

No.

**Comment by tdsmith** [comment]
With my data scientist hat on: I saw the comment you left in Google Docs about whether we should use separate events or a single event; I think the approach you're using here is fine.

**Review by Amejia481 — APPROVED** [review]

**Pull request merged**